### PR TITLE
AddSource Wizard - support to select an application

### PR DESCRIPTION
**Pull request by rvsia**
**Description**

Add an option to select an application when creating the source.

**Screenshots**

image

image

image

application

@Hyperkid123 @terezanovotna @gtanzillo @martinpovolny

### CODE REVIEW

File: packages/sources/src/addSourceWizard/SourceAddModal.js [hunk 1]
```diff
@@ -70,6 +96,7 @@ SourceAddModal.propTypes = {
     onCancel: PropTypes.func.isRequired,
     onSubmit: PropTypes.func.isRequired,
     sourceTypes: PropTypes.array,
+    applicationTypes: PropTypes.array,
```
Comment (Hyperkid123) [inline packages/sources/src/addSourceWizard/SourceAddModal.js]: I think we should add some shape for the array items.

File: packages/sources/src/addSourceWizard/SourceAddSchema.js [hunk 2]
```diff
@@ -317,14 +322,31 @@ const summaryStep = (sourceTypes) => ({
...
+    stepKey: 'application-type',
+    name: 'application-type',
+    title: 'Select application',
+    nextStep: 'summary',
+    fields: [
+        {
+            component: 'card-select',
+            name: 'app_type',
+            label: 'Select your application',
+            DefaultIcon: () => <></>,
```
Comment (Hyperkid123) [inline packages/sources/src/addSourceWizard/SourceAddSchema.js]: Use just Fragment please

File: packages/sources/src/api/index.js [hunk 3]
```diff
@@ -103,13 +118,14 @@ export function doCreateSource(formData, sourceTypes) {
...
-            console.error('Endpoint creation failure.');
-            throw { error: 'Endpoint creation failure.' };
+        }).catch(e => {
+            console.error(e);
+            throw { error: e };
         });
-
     }, (_error) => {
         console.error('Source creation failure.');
         throw { error: 'Source creation failure.' };
```
Comment (Hyperkid123) [inline packages/sources/src/api/index.js]: throw the actual error please.

File: packages/sources/src/sourceFormRenderer/components/SourceWizardSummary.js [hunk 4]
```diff
@@ -41,13 +43,16 @@ const SourceWizardSummary = ({ sourceTypes, formOptions }) => {
...
+            <TextListItem component={ TextListItemVariants.dt }>{ 'Application' }</TextListItem>
+            <TextListItem component={ TextListItemVariants.dd }>{ applicationName }</TextListItem>
         </TextList>
     </TextContent>);
 };
 
 SourceWizardSummary.propTypes = {
     formOptions: PropTypes.any.isRequired,
-    sourceTypes: PropTypes.arrayOf(PropTypes.any).isRequired
+    sourceTypes: PropTypes.arrayOf(PropTypes.any).isRequired,
```
Comment (Hyperkid123) [inline packages/sources/src/sourceFormRenderer/components/SourceWizardSummary.js]: Again we should have some shape here.

**Comment by rvsia** [comment]
@Hyperkid123 Fixed :+1:

**Comment by martinpovolny** [comment]
Looks great!

**Comment by rvsia** [comment]
@Hyperkid123 @karelhala Could you please release it (if it hasnt already been released)? 🙏

**Comment by Hyperkid123** [comment]
There you go: https://www.npmjs.com/package/@redhat-cloud-services/frontend-components-sources/v/0.0.5

**Pull request merged**